src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -153,6 +155,14 @@ func (s *fileSystemSource) Read(
 	}
 	builder := result.NewIndexBuilder(segBuilder)
 
+	// Preload info file results so they can be used to bootstrap data filesets and data migrations
+	infoFilesByNamespace := s.loadInfoFiles(namespaces)
+
+	// Perform any necessary migrations but don't block bootstrap process on failure. Will update info file
+	// in-memory structures in place if migrations have written new files to disk. This saves us the need from
+	// having to re-read migrated info files.
```

Comment:
If, for whatever reason, a migration consistently fails, will it get rerun every time that block is bootstrapped?

Reply:
Yeah. The mitigation would be disabling migrations for this version via the config. I think this is sufficient as anything more would probably require us to store state somewhere which I'm not sure is something we'd want to do

Reply:
Main concern is if there's a reproducible panic that will always fire during migration; any node attempting it will be stuck crash looping forever... I guess you could delete the fs data on that node and the new node will get peer streamed the migrated data, but that's a pretty heavy procedure operationally

Reply:
We can add the ability to recover from any panics caused by the migrator. That's probably the right thing to do anyway given that we don't ever want failed migrations to stop bootstrapping. 

I'm waffling on adding state to track migration failures, though. On one hand, it would be nice to see a certain number of failures and to automatically shut off the migrator. On the other hand, how we track it requires a little bit of consideration (e.g. fileset level vs global) and any mitigation is probably going to be manual anyway (e.g. deleting of filesets, filing a bug, etc) so it's probably not much more work to simply disable migrations in the config (especially considering that they opted-in in the first place). Interested in your thoughts here.